spectator-api/src/main/java/com/netflix/spectator/api/AbstractRegistry.java

```diff
@@ -221,4 +223,25 @@ private Meter compute(Meter m, Meter fallback) {
     PolledMeter.update(this);
     return meters.values().iterator();
   }
+
+  /**
+   * Can be called by sub-classes to remove expired meters from the internal map. It will
+   * look for meters that implement {@link SwapMeter} and set the implementation to null.
+   * If user code still holds a reference to the meter, then it should retrieve a new instance
+   * from the registry on the next access.
+   */
+  protected void removeExpiredMeters() {
```

Comment:
@brharrington Could you consider to make this method public?

In our use-case, we have our own Reporter and Scheduler (rather than use built into AtlasRegistry). So, we would like AtlasRegistry can removeExpiredMeters on our request, based on our schedule.